### PR TITLE
fix: remove await to send email async

### LINKED ISSUE #63: Remove all `await` keywords from infront of mailing functions
## Descriptions

image
We want the mails to be sent asynchronously. Adding an `await` in front of mail function blocks the thread. Remove all such occurences of `await`.

### PR DESCRIPTION
**Pull request by lakshay-saini-au8**
## Type
bug_fix

___

## Description
This PR focuses on making the email sending process asynchronous by removing the `await` keyword from the `sendOtp` and `sendEmail` function calls. This change is made in two files:
- `apps/api/src/auth/auth.service.ts`
- `apps/api/src/resend/services/mail.resend.ts`

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>auth.service.ts&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        apps/api/src/auth/auth.service.ts<br><br>

**The `await` keyword has been removed from the `sendOtp` <br>function call to make the email sending process <br>asynchronous.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/70/files#diff-7e9ac17d7e6417f73125395406cff59ad6eee28fab0bcca2993d72e2e439fae6"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mail.resend.ts&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        apps/api/src/resend/services/mail.resend.ts<br><br>

**The `await` keyword has been removed from the `sendEmail` <br>function call to make the email sending process <br>asynchronous.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/70/files#diff-ce34d66b2dee167e082fd748f86244ec90eb9a708474672d7e33dbd3c6329a2f"> +1/-1</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

## Description

Remove await keywords while sending email so that we can send it async.

Fixes #63 

## Dependencies

_Mention any dependencies/packages used_

## Future Improvements

_Mention any improvements to be done in future related to any file/feature_

## Mentions

@rajdip-b 

## Screenshots of relevant screens

_Add screenshots of relevant screens_

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

**Comment by rajdip-b** [comment]
I'll review this PR first thing in the morning tomorrow. Logged off just now :/

File: apps/api/src/auth/auth.service.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export class AuthService {
       this.OTP_EXPIRY
     )
 
-    await this.resend.sendOtp(email, otp.code)
+    this.resend.sendOtp(email, otp.code)
```
Comment (rajdip-b) [inline apps/api/src/auth/auth.service.ts:51]: The file is now situated under `src/auth/service` folder. I forgot to make the push yesterday, so please make sure to do this change

File: apps/api/src/resend/services/mail.resend.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ export class MailResend implements IResendService {
         </body>
         </html>
         `
-    await this.sendEmail(email, subject, body)
+    this.sendEmail(email, subject, body)
```
Comment (rajdip-b) [inline apps/api/src/resend/services/mail.resend.ts:90]: Lines 39 and 67 also contain the same thing, you can remove those aswell

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Comment by rajdip-b** [comment]
For resolving the merge conflict, you can just remove the `suth.service.ts` file inside the root of `auth` module

**Comment by kriptonian1** [comment]
@rajdip-b why are we taking contri on resend, haven't we planed to went with nodemailer

**Comment by rajdip-b** [comment]
> @rajdip-b why are we taking contri on resend, haven't we planed to go with `nodemailer`

this isnt specific to resend. the only change nodemailer integration will do has got nothing to do with doing this, apart from renaming the variables. so i guess we will be cool

**Comment by rajdip-b** [comment]
Hey @lakshay-saini-au8 I'm closing this PR for now, we want to integrate nodemailer first before we make any other changes since that is breaking things. Once it's done, I'll ping you so that you can take up the issue again!

**Pull request closed without merging**